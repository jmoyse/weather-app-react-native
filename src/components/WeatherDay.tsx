import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import WeatherIcon from './WeatherIcon';

interface WeatherDayProps {
    dayJSON: Object;
}

interface WeatherDayState { }

export default class WeatherDay extends React.Component<WeatherDayProps, WeatherDayState> {
    constructor (props: WeatherDayProps) {
        super(props);

    }

    render () {
        return (
            <View style={styles.container}>
                <Text style={styles.dayText} >
                    {(this.props.dayJSON as any).day}
                </Text>

                <WeatherIcon weatherID={Number.parseInt((this.props.dayJSON as any).code)} />

                <View style={styles.tempsView}>
                    <Text style={styles.highText} >
                        {(this.props.dayJSON as any).high + '°'}
                    </Text>
                    <Text style={styles.lowText} >
                        {(this.props.dayJSON as any).low + '°'}
                    </Text>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    lowText: {
        fontFamily: 'HelveticaNeueLTStd_Th',
        fontSize: 15,
        color: '#8FBFE8'
    },
    highText: {
        fontFamily: 'HelveticaNeueLTStd_Th',
        fontSize: 15,
        color: 'white',
        paddingRight: 15
    },
    dayText: {
        fontFamily: 'HelveticaNeueLTStd_Th',
        fontSize: 15,
        textAlign: 'center',
        color: 'white',
        width: 100
    },
    tempsView: {
        display: 'flex',
        flexDirection: 'row',
    },
    container: {
        //paddingTop: 5,
        //paddingLeft: 10,
        //paddingRight: 10,

        //height: 150,

        display: 'flex',
        flexDirection: 'row',
        //width: '100%',
        alignContent: 'center',
        alignItems: 'center',
        //alignItems: 'center',
        justifyContent: 'space-between',
    },
});
