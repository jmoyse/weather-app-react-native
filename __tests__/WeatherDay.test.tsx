import React from 'react';
import 'react-native';
import WeatherDay from '../src/components/WeatherDay';
import { shallow, configure, mount, ShallowWrapper, render, } from 'enzyme';

describe("WeatherDay", () => {
    let props = {
        dayJSON: {
            "code": "26",
            "date": "06 May 2018",
            "day": "Sun",
            "high": "33",
            "low": "29",
            "text": "Cloudy"
        }
    };
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<WeatherDay {...props} />);
    });

    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders text properly', () => {
        const dayText = wrapper.find('Text').get(0).props.children.toString();
        const highText = wrapper.find('Text').get(1).props.children.toString();
        const lowText = wrapper.find('Text').get(2).props.children.toString();

        expect(dayText).toEqual('Sunday');
        expect(highText).toEqual('33째');
        expect(lowText).toEqual('29째');
    });

    it('renders refreshes properly', () => {
        props.dayJSON.high = "34";
        props.dayJSON.low = "24";
        props.dayJSON.day = "Fri";

        wrapper.setProps({ ...props });

        const dayText = wrapper.find('Text').get(0).props.children.toString();
        const highText = wrapper.find('Text').get(1).props.children.toString();
        const lowText = wrapper.find('Text').get(2).props.children.toString();

        expect(dayText).toEqual('Friday');
        expect(highText).toEqual('34째');
        expect(lowText).toEqual('24째');
    });
});
