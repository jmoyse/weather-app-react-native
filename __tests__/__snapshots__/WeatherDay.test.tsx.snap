// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDay matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherDay
    dayJSON={
      Object {
        "code": "26",
        "date": "06 May 2018",
        "day": "Sun",
        "high": "33",
        "low": "29",
        "text": "Cloudy",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "Helvetica Neue",
              "fontSize": 15,
              "textAlign": "left",
              "width": 100,
            }
          }
        >
          Sunday
        </Text>,
        <Component
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "alignSelf": "center",
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <WeatherIcon
            weatherID={26}
          />
        </Component>,
        <Component
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontFamily": "Helvetica Neue",
                "fontSize": 15,
                "paddingRight": 15,
              }
            }
          >
            33°
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#8FBFE8",
                "fontFamily": "Helvetica Neue",
                "fontSize": 15,
              }
            }
          >
            29°
          </Text>
        </Component>,
      ],
      "style": Object {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 30,
        "justifyContent": "space-between",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Sunday",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "white",
            "fontFamily": "Helvetica Neue",
            "fontSize": 15,
            "textAlign": "left",
            "width": 100,
          },
        },
        "ref": null,
        "rendered": "Sunday",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WeatherIcon
            weatherID={26}
          />,
          "style": Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "position": "absolute",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "weatherID": 26,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 15,
                  "paddingRight": 15,
                }
              }
            >
              33°
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#8FBFE8",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 15,
                }
              }
            >
              29°
            </Text>,
          ],
          "style": Object {
            "display": "flex",
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "33°",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
                "fontFamily": "Helvetica Neue",
                "fontSize": 15,
                "paddingRight": 15,
              },
            },
            "ref": null,
            "rendered": "33°",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "29°",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#8FBFE8",
                "fontFamily": "Helvetica Neue",
                "fontSize": 15,
              },
            },
            "ref": null,
            "rendered": "29°",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontFamily": "Helvetica Neue",
                "fontSize": 15,
                "textAlign": "left",
                "width": 100,
              }
            }
          >
            Sunday
          </Text>,
          <Component
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "alignSelf": "center",
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <WeatherIcon
              weatherID={26}
            />
          </Component>,
          <Component
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 15,
                  "paddingRight": 15,
                }
              }
            >
              33°
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#8FBFE8",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 15,
                }
              }
            >
              29°
            </Text>
          </Component>,
        ],
        "style": Object {
          "alignContent": "center",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Sunday",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "white",
              "fontFamily": "Helvetica Neue",
              "fontSize": 15,
              "textAlign": "left",
              "width": 100,
            },
          },
          "ref": null,
          "rendered": "Sunday",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WeatherIcon
              weatherID={26}
            />,
            "style": Object {
              "alignContent": "center",
              "alignItems": "center",
              "alignSelf": "center",
              "position": "absolute",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "weatherID": 26,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 15,
                    "paddingRight": 15,
                  }
                }
              >
                33°
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#8FBFE8",
                    "fontFamily": "Helvetica Neue",
                    "fontSize": 15,
                  }
                }
              >
                29°
              </Text>,
            ],
            "style": Object {
              "display": "flex",
              "flexDirection": "row",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "33°",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 15,
                  "paddingRight": 15,
                },
              },
              "ref": null,
              "rendered": "33°",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "29°",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#8FBFE8",
                  "fontFamily": "Helvetica Neue",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "29°",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
